Name: Personal Avatar Left Side
Version: 1.3.5-4
Author: Elryon, dBsooner, Jason Robitaille
Description: This will enable a personal avatar on the left side of the conversation (MIT license)

--- a/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
+++ b/usr/palm/applications/com.palm.app.messaging/app/controllers/chatview-assistant.js
@@ -1409,7 +1409,7 @@ var ChatviewAssistant = Class.create({
 			if (ChatFlags.isOutgoing(msg.flags)) {
 				tmpClass.push("my-chat");
 				// Disabling MY avatar because we are not doing this for 1.0  2008-11-05
-				//msg.avatar = this.myAvatar || 'images/my-avatar.png';
+				msg.avatar = this.myAvatar || '/media/internal/my-avatar.png';
 			} else {
 				tmpClass.push("their-chat");
 				msg.avatar = this.avatarHash[msg.fromAddress] || this.contactAvatar || "images/list-view-default-avatar.png";
diff --git a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
index 73af153..68e74a4 100644
--- a/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
+++ b/usr/palm/applications/com.palm.app.messaging/stylesheets/messaging.css
@@ -613,6 +613,12 @@ img.avatar-overlay {
 .chat-entry .chat-balloon {
 	min-height: 26px;
 }                  
+.their-chat .message-container {
+	width: 91%;
+}
+.my-chat .message-container {
+	width: 91%;
+}
 .chat-entry.my-chat.status-message {
 	font-style: italic;
 }
@@ -622,14 +628,18 @@ img.avatar-overlay {
 	font-size:14px;
 }
 
-.my-chat .chat-balloon-wrapper, 
+.my-chat .chat-balloon-wrapper {
+	padding-left: 40px;
+	font-size: 18px;
+	word-wrap: break-word !important;
+}
 .their-chat .chat-balloon-wrapper {
   padding: 1px 0;
   font-size: 18px;
   word-wrap: break-word !important;
 }
 .their-chat .chat-balloon-wrapper {
-	padding-right: 8px;
+	padding-right: 20px;
 }
 .their-chat .avatar {
     width: 40px;
@@ -637,10 +647,20 @@ img.avatar-overlay {
 }
 
 /* Disabling MY avatar because we are not doing this for 1.0  2008-11-05*/
-.my-chat .avatar-location,
-.my-chat .avatar-overlay,
+.my-chat .avatar-location {
+	left: 12px;
+	position: absolute;
+	margin: 13px 12px 0px 0px;
+}
+.my-chat .avatar-overlay {
+	left: 8px;
+	margin-top: 4px;
+	position: absolute;
+}
 .my-chat .avatar-container {
-	display: none;
+	display: inline-block;
+}
+.my-chat .avatar {
     width: 40px;
     height: 40px;
 }
